llvm/lib/Analysis/MLInlineAdvisor.cpp

```diff
@@ -192,18 +193,27 @@ MLInlineAdvisor::MLInlineAdvisor(
   }
   NodeCount = AllNodes.size();
 
-  if (auto IR2VecVocabResult = MAM.getCachedResult<IR2VecVocabAnalysis>(M)) {
+  if (auto *IR2VecVocabResult = MAM.getCachedResult<IR2VecVocabAnalysis>(M)) {
     if (!IR2VecVocabResult->isValid()) {
       M.getContext().emitError("IR2VecVocabAnalysis is not valid");
       return;
     }
     // Add the IR2Vec features to the feature map
     auto IR2VecDim = IR2VecVocabResult->getDimension();
-    getFeatureMap().push_back(
+    FeatureMap.push_back(
         TensorSpec::createSpec<float>("callee_embedding", {IR2VecDim}));
-    getFeatureMap().push_back(
+    FeatureMap.push_back(
         TensorSpec::createSpec<float>("caller_embedding", {IR2VecDim}));
   }
+  if (InteractiveIncludeDefault)
```

Comment:
Should it also check for `!InteractiveChannelBaseName.empty()`?

Reply:
ya, but if not, and the model doesn't have it, it's a noop (minus compiletime penalty).